cloud/pod_agent_util.go

```diff
@@ -2,113 +2,50 @@ package cloud
 
 import (
 	"fmt"
-	"path/filepath"
 	"strings"
 
 	"github.com/evergreen-ci/evergreen"
 	"github.com/evergreen-ci/evergreen/model/pod"
 )
 
-// agentScript returns the script to provision and run the agent in the pod's
-// container.
-func agentScript(settings *evergreen.Settings, p *pod.Pod) []string {
-	scriptCmds := []string{downloadAgentCommands(settings, p)}
-	if p.TaskContainerCreationOpts.OS == pod.OSLinux {
-		scriptCmds = append(scriptCmds, fmt.Sprintf("chmod +x %s", clientName(p)))
-	}
-	agentCmd := strings.Join(agentCommand(settings, p), " ")
-	scriptCmds = append(scriptCmds, agentCmd)
+// bootstrapContainerCommand returns the script to bootstrap the pod's primary container
+// to run the agent.
+func bootstrapContainerCommand(settings *evergreen.Settings, p *pod.Pod) []string {
+	scriptCmds := downloadPodProvisioningScriptCommand(settings, p)
 
-	return append(invokeShellScriptCommand(p), strings.Join(scriptCmds, " && "))
+	return append(invokeShellScriptCommand(p), scriptCmds)
 }
 
 // invokeShellScriptCommand returns the arguments to invoke an in-line shell
 // script in the pod's container.
 func invokeShellScriptCommand(p *pod.Pod) []string {
 	if p.TaskContainerCreationOpts.OS == pod.OSWindows {
-		return []string{"cmd.exe", "/c"}
+		return []string{"powershell.exe", "-noninteractive", "-noprofile", "-Command"}
 	}
 
 	return []string{"bash", "-c"}
 }
 
-// agentCommand returns the arguments to start the agent in the pod's container.
-func agentCommand(settings *evergreen.Settings, p *pod.Pod) []string {
-	var pathSep string
-	if p.TaskContainerCreationOpts.OS == pod.OSWindows {
-		pathSep = "\\"
-	} else {
-		pathSep = "/"
-	}
-
-	return []string{
-		fmt.Sprintf(".%s%s", pathSep, clientName(p)),
-		"agent",
-		fmt.Sprintf("--api_server=%s", settings.ApiUrl),
-		"--mode=pod",
-		fmt.Sprintf("--log_prefix=%s", filepath.Join(p.TaskContainerCreationOpts.WorkingDir, "agent")),
-		fmt.Sprintf("--working_directory=%s", p.TaskContainerCreationOpts.WorkingDir),
-	}
-}
-
-// downloadAgentCommands returns the commands to download the agent in the pod's
-// container.
-func downloadAgentCommands(settings *evergreen.Settings, p *pod.Pod) string {
+func downloadPodProvisioningScriptCommand(settings *evergreen.Settings, p *pod.Pod) string {
```

Comment:
It is relatively straightforward but a comment would be nice to be consistent with the other util functions

Reply:
Added a comment.